projects/storefrontlib/shared/components/media/media.component.html

```diff
@@ -1,36 +1,77 @@
 <ng-container *ngIf="!!media">
-  <picture *ngIf="media.srcset && !isLegacy; else legacyImgTmp">
-    <source
-      *ngFor="
-        let source of media!.srcset! | cxMediaSources;
-        trackBy: trackByMedia
+  <ng-container *cxFeature="'useMediaComponentWithConfigurableMediaQueries'">
```

Comment:
If the toggle `useMediaComponentWithConfigurableMediaQueries` controls the template and with the active toggle by default we introduce the use of `<img>` tag and `<picture>` will be used only when we explicitely select it via input `usePictureElement`, we will introduce regression bugs which have been fixed within CXSPA-4114. Due to the lack of `sizes` attribute, the width descriptor in `srcset` will not be taken into account, and consequently, `srcset` used together with `img` tag will have no effect and images will not be responsive (see: https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/srcset)

We need to decide whether we want to include support for `sizes` (and maybe density descriptor) in the scope of this PR or as a separate task - I believe it requires more research.

If not, the introduced feature toggle should not control the method used to retrieve media data - whether from extended media queries or not. The template could stay as it was, so no `<img>`, only `<picture>` element.

Reply:
It looks that In the case of the `<img>` tag, the `sizes` attribute, if not provided, has a fallback to `100vw`. Apart from that, it depends on the browser how it is interpreted:
- Firefox picks the proper image size on change
- Chrome picks the proper image for the state of the viewport on the refresh, then if an image with a higher resolution is loaded, it will be used even if we decrease the screen width